burr/core/application.py

```diff
@@ -584,6 +606,224 @@ async def arun(
+                )
+                buffer = []
+                for chunk in response:
+                    delta = chunk.choices[0].delta.content
+                    buffer.append(delta)
+                    # yield partial results
+                    yield {'response': delta}
+                full_response = ''.join(buffer)
+                # return the final result
+                return {'response': full_response}, state.update(response=full_response)
+
+        To use streaming_result, you pass in names of streaming actions (such as the one above) to the halt_after
+        parameter:
+
+        .. code-block:: python
+
+            application = ApplicationBuilder().with_actions(streaming_response=streaming_response)...build()
+            prompt = "Count to 100, with a comma between each number and no newlines. E.g., 1, 2, 3, ..."
+            output = application.stream_result(halt_after='streaming_response', inputs={"prompt": prompt})
+            for result in output:
+                print(result['response']) # one by one
+
+            result, state = output.get()
+            print(result) #  all at once
+
+        Note that if you have multiple halt_after conditions, you can use the ``.action`` attribute to get the action that
+        was run.
+
+        .. code-block:: python
+
+            application = ApplicationBuilder().with_actions(
+                streaming_response=streaming_response,
+                error=error # another function that outputs an error, streaming
+            )...build()
+            prompt = "Count to 100, with a comma between each number and no newlines. E.g., 1, 2, 3, ..."
+            output = application.stream_result(halt_after='streaming_response', inputs={"prompt": prompt})
+            color = "red" if output.action.name == "error" else "green"
+            for result in output:
+                print(format(result['response'], color)) # assumes that error and streaming_response both have the same output shape
+
+        .. code-block:: python
+
+            application = ApplicationBuilder().with_actions(
+                streaming_response=streaming_response,
+                error=non_streaming_error # a non-streaming function that outputs an error
+            )...build()
+            prompt = "Count to 100, with a comma between each number and no newlines. E.g., 1, 2, 3, ..."
+            action, streaming_result = application.stream_result(halt_after='streaming_response', inputs={"prompt": prompt})
+            color = "red" if action.name == "error" else "green"
+            if action.name == "streaming_response": # can also use the ``.streaming`` attribute of action
+                for result in output:
+                     print(format(result['response'], color)) # assumes that error and streaming_response both have the same output shape
+            else:
+                result, state = output.get()
+                print(format(result['response'], color))
```

Comment:
what's the difference between these two examples?

Shouldn't `stream_result` always return a tuple?